birdhouse/config/jupyterhub/jupyterhub_config.py.template

```diff
@@ -20,7 +22,22 @@ c.JupyterHub.db_url = '/persist/jupyterhub.sqlite'
 
 c.JupyterHub.template_paths = ['/custom_templates']
 
-c.JupyterHub.spawner_class = 'dockerspawner.DockerSpawner'
+class CustomDockerSpawner(DockerSpawner):
+
+    def start(self):
+        if(os.environ['MOUNT_IMAGE_SPECIFIC_NOTEBOOKS'] == 'true'):
+            host_dir = join(os.environ['JUPYTERHUB_USER_DATA_DIR'], 'tutorial-notebooks', self.user_options.get('image'))
+            image_dir = join('/notebook_dir/tutorial-notebooks', self.user_options.get('image'))
```

Comment:
Just some thoughts on this : Since notebooks from only one image is mounted at the same time. We do not need this neat structure :
```
/notebook_dir/tutorial-notebooks/common
                                 crim-eo
                                 crim-nlp
```
Since at most two folders will always exist. We could simply have :

```
/notebook_dir/tutorial-notebooks/common
                                 current-image-specific
```
But as of writing this, I'm not sure that having a single folder name having its content change based on the current image is really better.

Reply:
I think that code is the layout "inside" the Jupyter server.

On disk:
```
/data/tutorial-notebooks/common
                                 crim-eo
                                 crim-nlp
```

Reply:
Yes, you're right this is the structure on disk. My point was that since we would never see that layout (with all images) inside the Jupyter server (because only one image is choosen at a time + common), it is not required to use their name (no collision possible). But again using the same folder for every image could be misleading.

Reply:
Yes it is the structure on disk.
I also use the same folder name structure on the image for volume mounting, but I could easily change the name of the sub-folder of the specific image to a more generic name such as 'current-image-specific' if we want.
But, using the name of the image as the name of the subfolder is okay too I think. The user sees easily that the folder contains the notebooks related to his environment, such as crim-eo.

Reply:
Exact, let's keep that this way.